homeassistant/helpers/entity.py

```diff
@@ -120,6 +120,10 @@ def assumed_state(self):
         """Return True if unable to access real state of entity."""
         return False
 
+    def finalize_setup(self):
```

Comment:
I prefer to not add a method to get this done. Instead I would suggest a similar approach to the [Cast media player platform](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/media_player/cast.py#L27) and keep track of the IDs or hosts of the platform that have been added and not add them again.